odoo/addons/base/models/qweb.py

```diff
@@ -655,115 +332,211 @@ def _compile_node(self, el, options):
-                                ],
-                                keywords=[],
-                                starargs=None, kwargs=None
-                            )
-                        ]
-                    ),
-                    body=[
-                        self._append(ast.Str(u' ')),
-                        self._append(ast.Name(id='name', ctx=ast.Load())),
-                        self._append(ast.Str(u'="')),
-                        self._append(ast.Call(
-                            func=ast.Name(id='escape', ctx=ast.Load()),
-                            args=[ast.Call(
-                                func=ast.Name(id='to_text', ctx=ast.Load()),
-                                args=[ast.Name(id='value', ctx=ast.Load())], keywords=[],
-                                starargs=None, kwargs=None
-                            )], keywords=[],
-                            starargs=None, kwargs=None
-                        )),
-                        self._append(ast.Str(u'"')),
-                    ],
-                    orelse=[]
-                )],
-                orelse=[]
-            )
-        ]
+    def _compile_options(self, el, varname, options, indent):
+        """
+        compile t-options and add to the dict the t-options-xxx values
+        """
+        t_options = el.attrib.pop('t-options', None)
+        directive = ''
+        if t_options:
+            directive = f't-options="{t_options}"'
+
+        code = []
+        dict_arg = []
+        for key in list(el.attrib):
+            if key.startswith('t-options-'):
+                value = el.attrib.pop(key)
+                directive += f' {key}="{value}"'
+                option_name = key[10:]
+                dict_arg.append(f'"{self._compile_str(option_name)}":{self._compile_expr(value)}')
+
+        if t_options and dict_arg:
+            code.append(self._indent(f"{varname} = {{**dict({self._compile_expr(t_options)}), {', '.join(dict_arg)}}}", indent))
+        elif dict_arg:
+            code.append(self._indent(f"{varname} = {{{', '.join(dict_arg)}}}", indent))
+        elif t_options:
+            code.append(self._indent(f"{varname} = dict({self._compile_expr(t_options)})", indent))
+
+        if code:
+            return self._compile_hook_before_directive(el, directive, options, indent) + \
+                code + \
+                self._compile_hook_after_directive(el, directive, options, indent)
+
+        return code
+
+    def _compile_format(self, expr):
+        """ Parses the provided format string and compiles it to a single
+        expression ast, uses string concatenation via "+".
+        """
+        text = '"'
+        base_idx = 0
+        for m in _FORMAT_REGEX.finditer(expr):
+            literal = expr[base_idx:m.start()]
+            if literal:
+                text += self._compile_str(literal if isinstance(literal, str) else literal.decode('utf-8'))
```

Comment:
Just `assert isinstance(expr, str)` and ignore the bytes case. It likely hasn't been necessary in a very long time.